File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/zaap/TeleportDestinationsListMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait TeleportDestinationsListMessage extends MessageDeserializer {
  val messageId = 5960

  protected trait Message extends BaseMessage with Serializable {
    val messageId = TeleportDestinationsListMessage.messageId

    val teleporterType: Byte
    val mapIds: Seq[Int]
    val subAreaIds: Seq[Short]
    val costs: Seq[Short]

    def serialize(buf: Buffer) {
      buf.writeByte(teleporterType)
      buf.writeUShort(mapIds.length)
      for (entry <- mapIds) {
        buf.writeInt(entry)
      }
      buf.writeUShort(subAreaIds.length)
      for (entry <- subAreaIds) {
        buf.writeShort(entry)
      }
      buf.writeUShort(costs.length)
      for (entry <- costs) {
        buf.writeShort(entry)
      }
    }

  }

  protected def create(teleporterType: Byte, mapIds: Seq[Int], subAreaIds: Seq[Short], costs: Seq[Short]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val teleporterType = buf.readByte()
    val mapIds = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }
    val subAreaIds = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val costs = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }

    create(teleporterType, mapIds, subAreaIds, costs)
  }

}

object TeleportDestinationsListMessage extends TeleportDestinationsListMessage {
  case class MessageImpl(teleporterType: Byte, mapIds: Seq[Int], subAreaIds: Seq[Short], costs: Seq[Short]) extends Message
  type Target = MessageImpl

  override def create(teleporterType: Byte, mapIds: Seq[Int], subAreaIds: Seq[Short], costs: Seq[Short]) = MessageImpl(teleporterType, mapIds, subAreaIds, costs)

  def apply(teleporterType: Byte, mapIds: Seq[Int], subAreaIds: Seq[Short], costs: Seq[Short]) = create(teleporterType, mapIds, subAreaIds, costs)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/TeleportDestinationsListMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class TeleportDestinationsListMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5960;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte teleporterType;
    public int[] mapIds;
    public short[] subAreaIds;
    public short[] costs;
    
    public TeleportDestinationsListMessage() { }
    
    public TeleportDestinationsListMessage(byte teleporterType, int[] mapIds, short[] subAreaIds, short[] costs) {
        this.teleporterType = teleporterType;
        this.mapIds = mapIds;
        this.subAreaIds = subAreaIds;
        this.costs = costs;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(teleporterType);
        buf.writeUShort(mapIds.length);
        for (int entry : mapIds) {
            buf.writeInt(entry);
        }
        buf.writeUShort(subAreaIds.length);
        for (short entry : subAreaIds) {
            buf.writeShort(entry);
        }
        buf.writeUShort(costs.length);
        for (short entry : costs) {
            buf.writeShort(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        teleporterType = buf.readByte();
        if (teleporterType < 0)
            throw new RuntimeException("Forbidden value on teleporterType = " + teleporterType + ", it doesn't respect the following condition : teleporterType < 0");
        int limit = buf.readUShort();
        mapIds = new int[limit];
        for (int i = 0; i < limit; i++) {
            mapIds[i] = buf.readInt();
        }
        limit = buf.readUShort();
        subAreaIds = new short[limit];
        for (int i = 0; i < limit; i++) {
            subAreaIds[i] = buf.readShort();
        }
        limit = buf.readUShort();
        costs = new short[limit];
        for (int i = 0; i < limit; i++) {
            costs[i] = buf.readShort();
        }
    }
    
}
